discovery-provider/src/challenges/listen_streak_challenge.py

```diff
@@ -0,0 +1,72 @@
+from collections import Counter
+from datetime import datetime
+from src.challenges.challenge import ChallengeManager, ChallengeUpdater
+from src.models import ListenStreakChallenge, Play
+from src.challenges.challenge_event import ChallengeEvent
+
+class ListenStreakChallengeUpdater(ChallengeUpdater):
```

Comment:
let's add types to these methods if you don't mind :) I can help with your IDE setup if things aren't working nicely, types should give you better intellisense completion and add autocompletes, etc